apps/ui/src/lib/store/yjsSlice.ts

```diff
@@ -197,10 +197,18 @@ export const createYjsSlice: StateCreator<MyState, [], [], YjsSlice> = (
           });
         }
       );
-      // Set active runtime to the first one.
+      // Set active runtime
       const runtimeMap = get().getRuntimeMap();
       if (runtimeMap.size > 0) {
-        get().setActiveRuntime(Array.from(runtimeMap.keys())[0]);
+        const runtimeList = Array.from(runtimeMap.keys());
+        const activeInstances = runtimeList.filter((id) => {
+          return runtimeMap.get(id)?.status !== undefined;
```

Comment:
The status must `=== "connected"` to be alive. It could be undefined, "disconnected", or "connecting".

But I'd suggest keeping the original logic (use the first runtime regardless of its status) for now.

There are many remaining logic to add:
1. This assignment only happens at the initial page loading. When there's no runtime, creating a new runtime wouldn't activate it. So we also need to automatically activate the runtime when creating a new runtime (if there's no runtime).
2. We also need to create runtime automatically when creating or opening a repo.

Reply:
I will continue to work on the runtime-related staff. You can focus on the selective view and don't worry too much at the runtime.

Reply:
SG, I noticed that the active runtime continues reset in the UI refactor PR, and realized the root cause here. I will close this PR then.

Reply:
> active runtime continues rese

The runtime isn't stable, but this isn't the root cause, and fixing this alone couldn't make it stable.